src/components/pages/dashboard/dashboard.js

```diff
@@ -38,31 +45,79 @@ class DashboardPage extends Component {
     }
   }
 
-  shouldComponentUpdate(nextProps, nextState) {
-    return !_.isEqual(this.props, nextProps) || !_.isEqual(this.state, nextState );
+  loadMapData() {
+    const deviceIds = ((this.props.devices || {}).items || []).map(({Id}) => Id) || [];
+    this.props.actions.loadDashboardData(deviceIds, this.state.timeRange);
+  }
+
+  onTimeRangeChange(selectedOption) {
+    if (!selectedOption) return;
+    this.setState({
+      timeRange: selectedOption.value,
+      lastRefreshed: new Date()
+    },
+      this.loadMapData()
```

Comment:
This doesn't seem to be correct? does loadMapData return a function?

Reply:
it's trigger's load alarms and telemetry data for map